### ISSUE TITLE
Cannot build from source anymore

### ISSUE THREAD

**Issue by Th0masL** [OpenNebula/terraform-provider-opennebula#165]
It seems like we cannot build from the current master branch anymore (master/2fd53ed).

I've tested to build both on MacOS and Ubuntu20

## On MacOS

1. Try to build the provider
```
# go build -o /tmp/terraform-provider-opennebula
go: github.com/OpenNebula/one/src/oca/go/src/goca@v0.0.0-20210824074935-e55e3a53c4c0: missing go.sum entry; to add it:
	go mod download github.com/OpenNebula/one/src/oca/go/src/goca
go: github.com/OpenNebula/one/src/oca/go/src/goca@v0.0.0-20210824074935-e55e3a53c4c0: missing go.sum entry; to add it:
	go mod download github.com/OpenNebula/one/src/oca/go/src/goca
```

2. Install goca
```
# go mod download github.com/OpenNebula/one/src/oca/go/src/goca
```

3. Try to build the provider again
```
# go build -o /tmp/terraform-provider-opennebula
opennebula/helpers_vm.go:11:2: cannot find module providing package github.com/hashicorp/terraform/helper/schema: import lookup disabled by -mod=readonly
	(Go version in go.mod is 1.13, so vendor directory was not used.)
```

4. Try (fail) to install `terraform/helper/schema`
```
# go get -v github.com/hashicorp/terraform/helper/schema
go get: module github.com/hashicorp/terraform@upgrade found (v1.0.8), but does not contain package github.com/hashicorp/terraform/helper/schema
```

## On Ubuntu20

1. Try to build the provider
```
# root@server1:/tmp/git/terraform-provider-opennebula# go build -o terraform-provider-opennebula
go: downloading github.com/hashicorp/terraform-plugin-sdk v1.7.0
go: downloading github.com/OpenNebula/one/src/oca/go/src/goca v0.0.0-20210824074935-e55e3a53c4c0
<... output truncated ...>
go: finding github.com/hashicorp/terraform-config-inspect v0.0.0-20191115094559-17f92b0546e8
go: finding github.com/hashicorp/hcl v0.0.0-20170504190234-a4b07c25de5f
build github.com/OpenNebula/terraform-provider-opennebula: cannot load github.com/hashicorp/terraform/helper/schema: module github.com/hashicorp/terraform@latest found (v1.0.8), but does not contain package github.com/hashicorp/terraform/helper/schema
```

2. Try (fail) to install `terraform/helper/schema`
```
# root@server1:/tmp/git/terraform-provider-opennebula# go get -v github.com/hashicorp/terraform/helper/schema
go get github.com/hashicorp/terraform/helper/schema: module github.com/hashicorp/terraform@upgrade found (v1.0.8), but does not contain package github.com/hashicorp/terraform/helper/schema
```

Has there been some changes in the way to build the provider ?

**Comment by treywelsh** [comment]
Sorry for this obvious error on master branch, PR #166 should fix this

**Comment by treywelsh** [comment]
Merged into master

**Comment by Th0masL** [comment]
Thanks, I can now build from source.

The new version of the plugin successfully pass `terraform init` and `terraform validate`, but does not pass `terraform plan`.

Here is the error :
```
terraform plan -compact-warnings -input=false -no-color -target=module.test1 -out=/tmp/test1.tfplan
module.test1.module.varnish1[0].random_uuid.test: Refreshing state... [id=abb43071-c724-8886-7c90-c293470e0556]
module.test1.module.vmgroup.opennebula_virtual_machine_group.vm-group-server1: Refreshing state... [id=27]
module.test1.module.securitygroup.opennebula_security_group.internal: Refreshing state... [id=123]
module.test1.module.securitygroup.opennebula_security_group.server1: Refreshing state... [id=124]

Warning: Resource targeting is in effect

You are creating a plan with the -target option, which means that the result of this plan may not represent all of the changes requested by the current configuration.

The -target option is not for routine use, and is provided only for exceptional situations such as recovering from errors or mistakes, or when Terraform specifically suggests to use it as part of an error message.

Error: Plugin did not respond

  with module.test1.module.securitygroup.opennebula_security_group.server1,
  on modules/securitygroup/main.tf line 40, in resource "opennebula_security_group" "server1":
  40: resource "opennebula_security_group" "server1" {

The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.

Stack trace from the terraform-provider-opennebula plugin:

panic: interface conversion: interface {} is int, not string

goroutine 253 [running]:
github.com/OpenNebula/terraform-provider-opennebula/opennebula.matchNIC(0xc000559ecc, {{{{0x0, 0x0}, {0xc0004ae22b, 0x3}}, {0xc000242c00, 0xf, 0x10}}})
	/Users/thomas/github/OpenNebula/terraform-provider-opennebula/opennebula/resource_opennebula_virtual_machine.go:765 +0x694
github.com/OpenNebula/terraform-provider-opennebula/opennebula.flattenVMNIC(0xc000306230, 0x1928d88)
	/Users/thomas/github/OpenNebula/terraform-provider-opennebula/opennebula/resource_opennebula_virtual_machine.go:811 +0x512
github.com/OpenNebula/terraform-provider-opennebula/opennebula.resourceOpennebulaVirtualMachineReadCustom(0x1901f40, {0x19182e0, 0xc0002e33e0}, 0x196d640, 0x196d648)
	/Users/thomas/github/OpenNebula/terraform-provider-opennebula/opennebula/resource_opennebula_virtual_machine.go:510 +0x3b6
github.com/OpenNebula/terraform-provider-opennebula/opennebula.resourceOpennebulaVirtualMachineRead(...)
	/Users/thomas/github/OpenNebula/terraform-provider-opennebula/opennebula/resource_opennebula_virtual_machine.go:533
github.com/OpenNebula/terraform-provider-opennebula/opennebula.resourceOpennebulaVirtualMachineExists(0xc000306230, {0x19182e0, 0xc0002e33e0})
	/Users/thomas/github/OpenNebula/terraform-provider-opennebula/opennebula/resource_opennebula_virtual_machine.go:875 +0x3a
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00052e600, 0xc0005140a0, {0x19182e0, 0xc0002e33e0})
	/Users/thomas/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.7.0/helper/schema/resource.go:440 +0x131
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000010ae8, {0xc0004eed70, 0x108ebc6}, 0xc0004eed70)
	/Users/thomas/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.7.0/internal/helper/plugin/grpc_provider.go:525 +0x34b
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler({0x18f82a0, 0xc000010ae8}, {0x1a9e130, 0xc00064e210}, 0xc00068f260, 0x0)
	/Users/thomas/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.7.0/internal/tfplugin5/tfplugin5.pb.go:3153 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000d14a0, {0x1aab008, 0xc000188180}, 0xc0006a4200, 0xc00054cff0, 0x2005b70, 0x0)
	/Users/thomas/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:995 +0xe0d
google.golang.org/grpc.(*Server).handleStream(0xc0000d14a0, {0x1aab008, 0xc000188180}, 0xc0006a4200, 0x0)
	/Users/thomas/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1275 +0xa56
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	/Users/thomas/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:710 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/thomas/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:708 +0xef

Error: The terraform-provider-opennebula plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

One thing that looks strange is that terraform is showing the path of where I have cloned the github repo (`/Users/thomas/github/OpenNebula/terraform-provider-opennebula`), but I'm not running terraform commands in this folder.

**Comment by treywelsh** [comment]
It's due to some type conversion errors introduced by #143.
I didn't reproduced it for now but here is a quick fix: https://github.com/treywelsh/terraform-provider-opennebula/tree/B-165
However acceptance tests may still fail, I'll work on it and propose a PR as soon as possible

By the way, thank you for your error reports

**Comment by treywelsh** [comment]
I made a bit more generic issue to track recent master branch problems: #167 
And a related PR to fix all at once:  #168

Feel free to test the PR branch, as usual all kind of feedback would be appreciated

I close this issue
